lib/AST/DiagnosticEngine.cpp

```diff
@@ -110,6 +110,12 @@ static constexpr const char *const debugDiagnosticStrings[] = {
     "<not a diagnostic>",
 };
 
+static constexpr const char *const diagnosticIDStrings[] = {
+#define DIAG(KIND, ID, Options, Text, Signature) #ID,
+#include "swift/AST/DiagnosticsAll.def"
+    "",
```

Comment:
Most of the other tables like this use `"<not a diagnostic>"`.  Should we match that?